### ISSUE TITLE
[clsim] I3CLSimServer deadlocks if given more than 1 I3CLSimStepToPhotonConverter (Trac #2360)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2352]
I3CLSimServer appears to deadlock if trying to service more than GPU at a time. @lulu observed this behavior when attempting to run hobo-snowstorm on interactive multi-GPU nodes in Chiba.

This can also be reproduced by running e.g. 

```text
CUDA_VISIBLE_DEVICES=0,1 ./env-shell.sh clsim/resources/scripts/benchmark.py -n 1
```
on a dual-GPU system. With either `CUDA_VISIBLE_DEVICES=0` or `CUDA_VISIBLE_DEVICES=1`, it completes within a few seconds. With both GPUs enabled, it hangs forever.

Migrated from https://code.icecube.wisc.edu/ticket/2360
```json
{
    "status": "closed", 
    "changetime": "2019-12-05T20:04:42", 
    "description": "I3CLSimServer appears to deadlock if trying to service more than GPU at a time. @lulu observed this behavior when attempting to run hobo-snowstorm on interactive multi-GPU nodes in Chiba.\n\nThis can also be reproduced by running e.g. \n{{{\nCUDA_VISIBLE_DEVICES=0,1 ./env-shell.sh clsim/resources/scripts/benchmark.py -n 1\n}}}\non a dual-GPU system. With either `CUDA_VISIBLE_DEVICES=0` or `CUDA_VISIBLE_DEVICES=1`, it completes within a few seconds. With both GPUs enabled, it hangs forever.", 
    "reporter": "jvansanten", 
    "cc": "eganster, lulu", 
    "resolution": "fixed", 
    "_ts": "1575576282029750", 
    "component": "combo simulation", 
    "summary": "[clsim] I3CLSimServer deadlocks if given more than 1 I3CLSimStepToPhotonConverter", 
    "priority": "major", 
    "keywords": "", 
    "time": "2019-09-20T02:42:58", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20190923T16:25:21: olivas changed component from "analysis" to "simulation"

**Comment by icecube-trac** [comment]
Trac update at 20191205T20:04:42: 
* jvansanten changed resolution from "" to "fixed"
* jvansanten commented:

> In changeset 177702:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="177702"
> Use multiple devices in I3CLSimServer
> 
> The barrier logic assumed that bunches would return in order, which
> stopped being true as soon as there were multiple devices.
> 
> Fixes #2352.
> ```


* jvansanten changed status from "new" to "closed"
